### DISCUSSION TITLE
vite-node force ESM?

### DISCUSSION THREAD

**Discussion by bestickley** [vitest-dev/vitest#2909] [Q&A] [answered]
Hi, I'm trying to use `vite-node` as a replacement for `ts-node`. However, I've run into this issue:

```bash
vite-node ~/Code/green-boost/packages/gboost/src/index.ts create
(node:57507) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/Users/stickb/Code/green-boost/node_modules/.pnpm/@aws-sdk+util-user-agent-node@3.272.0/node_modules/@aws-sdk/util-user-agent-node/dist-es/is-crt-available.js:1
export const isCrtAvailable = () => {
^^^^^^

SyntaxError: Unexpected token 'export'
    at internalCompileFunction (node:internal/vm:74:18)
    at wrapSafe (node:internal/modules/cjs/loader:1141:20)
    at Module._compile (node:internal/modules/cjs/loader:1182:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1272:10)
    at Module.load (node:internal/modules/cjs/loader:1081:32)
    at Function.Module._load (node:internal/modules/cjs/loader:922:12)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:169:29)
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.13.0
```

This makes sense to me if vite-node is assuming a CJS environment, but I'm not sure why it would be. How does vite-node determine which module resolution to do? The closest package.json is `"type":  "module"`.

**Comment by sheremet-va** [comment, marked as answer]
This is Node.js error, not vite-node error. Vite-node just imports the file. The closes package.jsom doesn't have `"type": "module"`.

**Reply by bestickley** [reply]
Thank you for the quick reply @sheremet-va. Is there anywhere I can find docs on how the closest package.json is found? There exists a package.json at ~/Code/green-boost/packages/gboost/package.json with `"type": "module"` but does it look elsewhere? I tried running: `vite-node --root ~/Code/green-boost/packages/gboost ~/Code/green-boost/packages/gboost/src/index.ts create` but same error :/

**Reply by sheremet-va** [reply]
Closes package.json is the closest package.json to the file that gets imported. In your case closes package.json would be `/Users/stickb/Code/green-boost/node_modules/.pnpm/@aws-sdk+util-user-agent-node@3.272.0/node_modules/@aws-sdk/util-user-agent-node/package.json`

**Reply by bestickley** [reply]
@sheremet-va, understood. thank you for explaining! I'm tracking this issue here: https://github.com/aws/aws-sdk-js-v3/issues/3622

**Reply by bestickley** [reply]
Work around can be found here: https://github.com/vitest-dev/vitest/issues/2579
